### PR TITLE
Exposing ALLOW_AUTHZ

### PR DESCRIPTION
**Pull request by zxan1285**
##✏️ Changes
As `wt_edit` is no longer works, and users need to be able to use authz-extension integration, if they want to, we're adding `ALLOW_AUTHZ` secret to the `webtask.json`.

##🎯 Testing
✅ This has been tested locally
✅ This has been tested in a webtask
🚫 This change does not have unit test coverage

### CODE REVIEW

**Comment by auth0extensions** [comment]
MD5: 36df9540a5ef4996a9737657e4a8929c

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **14**
Affected Modules: **hoek, hoek, hoek, hoek, superagent, superagent, superagent, mime, mime, mime, superagent, superagent, superagent, lodash**
Summary:

- Prototype pollution attack
- Prototype pollution attack
- Prototype pollution attack
- Prototype pollution attack
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Prototype Pollution


## Detailed report
#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-express-tools@1.1.7 > webtask-tools@3.2.1 > boom@2.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-tools@1.3.2 > webtask-tools@3.2.1 > boom@2.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-sso-dashboard@2.0.1 > jsonwebtoken@7.4.3 > joi@6.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-sso-dashboard@2.0.1 > webtask-tools@3.2.1 > boom@2.10.1 > hoek@2.16.3**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-express-tools@1.1.7 > webtask-tools@3.2.1 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-tools@1.3.2 > webtask-tools@3.2.1 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > webtask-tools@3.2.1 > superagent@1.8.5**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-express-tools@1.1.7 > webtask-tools@3.2.1 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-tools@1.3.2 > webtask-tools@3.2.1 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-sso-dashboard@2.0.1 > webtask-tools@3.2.1 > superagent@1.8.5 > mime@1.3.4**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-express-tools@1.1.7 > auth0-extension-tools@1.3.2 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > auth0-extension-tools@1.3.2 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-sso-dashboard@2.0.1 > supertest@2.0.1 > superagent@2.3.0**

---

#### Prototype Pollution

Affected Module: **lodash**
Installed version: **4.5.1**
Patched versions: **>=4.17.5**
Advisory: https://nodesecurity.io/advisories/577
Path: **auth0-sso-dashboard@2.0.1 > lru-memoizer@1.10.0 > lodash@4.5.1**

---

File: webtask.json [hunk 1]
```diff
@@ -36,6 +36,22 @@
       "description": "A CSS file containing custom styles for the extension",
       "example": "https://cdn.fabrikam.com/static/extensions/theme/fabrikam.css",
       "required": false
+    },
+    "ALLOW_AUTHZ": {
+      "description": "This setting allow to enable Authorization Extension integration (Deprecated)",
```
Comment (fyockm) [inline webtask.json]: Let's just shorten this description to `Enable Authorization Extension integration (Deprecated)`

Comment (zxan1285) [reply webtask.json]: That's not quite true. `ALLOW_AUTHZ` **allows** user to enable integration. To actually enable it, user should go to settings in the extension itself, and enable it there.

File: server/lib/config.js [hunk 2]
```diff
@@ -1,16 +1,24 @@
 const settings = { };
 let currentProvider = null;
 
+const boolify = (item) => {
+  if (item === 'true' || item === 'false') {
+    return item === 'true';
```
Comment (fyockm) [inline server/lib/config.js]: I'm confused by this logic. why isn't it just this?

```
const boolify = (item) => item === 'true' || item === true;
```

File: webtask.json (hunk 1 above)
Comment (fyockm) [reply webtask.json]: Ok, let's make it `Allow enabling of Authorization Extension integration (Deprecated)`

File: server/lib/config.js (hunk 2 above)
Comment (fyockm) [reply server/lib/config.js]: we're trying to `boolify` everything,  so if we're trying to boolify `AUTH0_CLIENT_ID`, we should get client_id and not `false`

**Comment by zxan1285** [comment]
Fixed.

**Comment by saltukalakus** [comment]
Hi @zxan1285 ,  a quick question, which this PR, the Issues https://github.com/auth0-extensions/auth0-sso-dashboard-extension/issues/7 will be resolved. Is that right?

**Comment by zxan1285** [comment]
> Hi @zxan1285 , a quick question, with this PR, the Issues #7 will be resolved. Is that right?
Yes, this PR is making authz-integration available for customers. The feature was added some time ago, but it was disabled.

**Pull request merged**